src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java

```diff
@@ -163,13 +165,25 @@ public void enterCodesystemDefinition(cqlParser.CodesystemDefinitionContext ctx)
       CQLCodeSystem codeSystem = new CQLCodeSystem();
       codeSystem.setId(csDef.getId());
       codeSystem.setOID(csDef.getId());
-      codeSystem.setCodeSystemVersion(csDef.getVersion());
+      codeSystem.setCodeSystemVersion(getParsedVersion(csDef.getVersion()));
       codeSystem.setCodeSystemName(csDef.getName());
 
       codeSystemMap.putIfAbsent(identifier, codeSystem);
     }
   }
 
+  private String getParsedVersion(String version){
+    if(version != null) {
+      String versionRegex= "urn:hl7:version:(\\d+(\\.\\d+)?)";
```

Comment:
Will this capture versions containing hyphens or multiple periods, e.g. SNOMED's 2024-03 and ActCode's 9.0.0?

Reply:
An alternative approach is to use substring since the version prefix is a constant and the value always appears in the same location:
```csDef.getVersion().substring("urn:hl7:version:".length())```

Reply:
Thank you, I just pushed the code with the changes